modules/indexing/src/main/java/org/apache/ignite/internal/processors/query/h2/database/H2Tree.java

```diff
@@ -252,6 +255,21 @@ public H2Tree(
             // Page is ready - read meta information.
             MetaPageInfo metaInfo = getMetaInfo();
 
+            IgniteProductVersion ver_8_7_40 = IgniteProductVersion.fromString("8.7.40");
+            IgniteProductVersion ver_8_8_11 = IgniteProductVersion.fromString("8.8.11");
+
+            IgniteProductVersion idxCreateVer = metaInfo.createdVersion();
+
+            if (idxCreateVer != null) {
+                if ((idxCreateVer.compareTo(ver_8_7_40) >= 0 && idxCreateVer.minor() == 7) ||
+                    idxCreateVer.compareTo(ver_8_8_11) >= 0)
+                    useFixedComparator = true;
+                else
+                    useFixedComparator = false;
+            }
+            else
+                useFixedComparator = false;
```

Comment:
```suggestion
            useFixedComparator = idxCreateVer != null &&(idxCreateVer.compareTo(ver_8_7_40) >= 0 && idxCreateVer.minor() == 7) ||
                    idxCreateVer.compareTo(ver_8_8_11) >= 0);                   
```

Reply:
I found my approach is more readable as for me, i wan`t fix it ..

Reply:
`useFixedComparator = false` is a default value, thus similar 'else' branches looks too verbose.
What about the next way?
```
 if (idxCreateVer == null /* too old version */ ||
     idxCreateVer.compareTo(ver_8_7_40) < 0 || 
     idxCreateVer.minor() != 7 && idxCreateVer.compareTo(ver_8_8_11) < 0)
            useFixedComparator = false; // Fallback to legacy comparator due to compatibility reasons.
else 
     useFixedComparator = true;
```

Reply:
done thanks